sdk/go/pulumi/rpc.go

```diff
@@ -121,11 +124,17 @@ func marshalInputs(props Input) (resource.PropertyMap, map[string][]URN, []URN,
 		if err != nil {
 			return err
 		}
-		deps.union(allDeps)
+		for k := range allDeps {
+			deps.add(k)
+		}
 
 		if !v.IsNull() || len(allDeps) > 0 {
+			urns := slice.Prealloc[URN](len(allDeps))
+			for v := range allDeps {
+				urns = append(urns, v)
+			}
 			pmap[resource.PropertyKey(pname)] = v
-			pdeps[pname] = allDeps.values()
+			pdeps[pname] = urns
```

Comment:
Any benefit to this change?

Reply:
`allDeps.values()` returned a nil array, but `maps.Keys(allDeps)` returns an empty map, so I am keeping that behaviour here.